nl.sarine.gpx-viewer.json

```diff
@@ -0,0 +1,102 @@
+    "command" : "gpx-viewer",
+    "rename-desktop-file" : "gpx-viewer.desktop",
+    "rename-icon" : "gpx-viewer",
+    "finish-args" : [
+        "--share=ipc",
+        "--socket=fallback-x11",
+        "--socket=wayland",
+        "--device=dri",
+        "--share=network",
+        "--env=G_ENABLE_DIAGNOSTICS=1"
+    ],
+    "cleanup" : [
+        "/include",
+        "/lib/pkgconfig",
+        "/share/pkgconfig",
+        "/share/aclocal",
+        "/man",
+        "/share/man",
+        "/share/gtk-doc",
+        "/share/vala",
+        "*.la",
+        "*.a"
+    ],
+    "modules" : [
+        {
+            "name": "libchamplain",
+            "buildsystem": "meson",
+            "config-opts": [
+                "-Dvapi=true"
+            ],
+            "sources": [
+                {
+                    "type": "archive",
+                    "url": "https://download.gnome.org/sources/libchamplain/0.12/libchamplain-0.12.20.tar.xz",
+                    "sha256": "0232b4bfcd130a1c5bda7b6aec266bf2d06e701e8093df1886f1e26bc1ba3066",
+                    "x-checker-data": {
+                        "type": "gnome",
+                        "name": "libchamplain"
+                    }
+                }
+            ]
+        },
+        {
+            "name": "intltool",
+            "cleanup": [ "*" ],
+            "sources": [
+                {
+                    "type": "archive",
+                    "url": "https://launchpad.net/intltool/trunk/0.51.0/+download/intltool-0.51.0.tar.gz",
+                    "sha256": "67c74d94196b153b774ab9f89b2fa6c6ba79352407037c8c14d5aeb334e959cd"
+                }
+            ]
+        },
+        {
+            "name": "gdl",
+            "cleanup" : [
+                "/include",
+                "/lib/pkgconfig"
+            ],
+            "sources": [
+                {
+                    "type": "archive",
+                    "url": "https://download.gnome.org/sources/gdl/3.34/gdl-3.34.0.tar.xz",
+                    "sha256": "858b30f0cdce4c4cb3e8365a7d54ce57c388beff38ea583be5449bc78dda8d02"
+                }
+            ]
+        },
+        {
+            "name": "metainfo",
+            "buildsystem": "simple",
+            "build-commands": [
+                "install -D -m0644 nl.sarine.gpx-viewer.metainfo.xml /app/share/metainfo/nl.sarine.gpx-viewer.metainfo.xml"
+            ],
+            "sources": [
+                {
+                    "type": "file",
+                    "path": "nl.sarine.gpx-viewer.metainfo.xml"
+                }
+            ]
+        },
+        {
+            "name" : "gpx-viewer",
+            "buildsystem" : "meson",
+            "builddir" : true,
+            "sources" : [
+                {
+                    "type" : "git",
+                    "url" : "https://github.com/DaveDavenport/gpx-viewer"
+                },
+                {
+                    "type" : "patch",
+                    "path" : "add-128-icon.patch"
```

Comment:
Set `use-git` property  to `true` for this patch source to use git binary diffs.

Reply:
Ah cool. I open a PR upstream too: https://github.com/DaveDavenport/gpx-viewer/pull/12